File: backend/src/main/java/com/online/giftshop/services/CartService.java
/**
 * @author Gagandeep Singh
 * @email singh.gagandeep3911@gmail.com
 * @create date 2021-01-13 22:59:25
 * @modify date 2021-01-13 22:59:25
 * @desc [description]
 */
package com.online.giftshop.services;

import com.online.giftshop.dto.CartDto;
import com.online.giftshop.dto.ItemDto;
import com.online.giftshop.entities.Cart;

public interface CartService {

  Cart fetchCartById(Long cartId);

  Cart addToCart(ItemDto itemDto, Long id);

  Cart deleteFromCart(Long productId, Long id);

  // Merge local cart with Server cart
  // Can be done later
  Cart mergeCart(CartDto cartDto,Long id);

  Cart fetchByUserId(Long userId);

}

File: frontend/src/app/services/cart.service.ts
/**
 * @author Gagandeep Singh
 * @email singh.gagandeep3911@gmail.com
 * @create date 2021-01-27 02:57:17
 * @modify date 2021-01-27 02:57:17
 * @desc [description]
 */
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from 'src/environments/environment';
import { Item } from '../models/item.model';
import { AuthService } from './auth.service';

@Injectable({
  providedIn: 'root'
})
export class CartService {

  cartServiceUrl = `${environment.protocol}${environment.applicationUrl}/cart`;

  constructor(private http: HttpClient, private authService: AuthService) { }

  addToCart(item: Item) {
    return this.http.put(`${this.cartServiceUrl}/${this.authService.fetchFromSessionStorage()?.userId}`, item);
  }

  fetchCartByUserId() {
    return this.http.get(`${this.cartServiceUrl}/user/${this.authService.fetchFromSessionStorage()?.userId}`);
  }

  removeItemFromCart(productId) {
    return this.http.delete(`${this.cartServiceUrl}/delete/${productId}/${this.authService.fetchFromSessionStorage()?.userId}`)
  }

  checkout(data) {
    return this.http.post(`${this.cartServiceUrl}/manualcheckout/${this.authService.fetchFromSessionStorage()?.userId}`, data)
  }
}
